src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10558,8 +10579,12 @@ def diff(self, periods: int, axis: int) -> "SnowflakeQueryCompiler":
         }
         return SnowflakeQueryCompiler(
             self._modin_frame.update_snowflake_quoted_identifiers_with_expressions(
-                diff_label_to_value_map,
-                self._modin_frame.cached_data_column_snowpark_pandas_types,
```

Comment:
In which case, that the result of diff has different types than the original df?

Reply:
diff() between two timestamp columns on axis=1, or on a timestamp column on axis 0, will change type to timedelta. I added a comment.